File: c/unique_character.c
#include <stdio.h>

int hasAllUniqueCharacters(const char *str) {
    int checker = 0;
    while (*str) {
        if (*str < 'a' || *str > 'z') {
            fprintf(stderr, "Character '%c' is not a lowercase letter (a-z).\n", *str);
            return -1;
        }
        int bitIndex = *str - 'a';
        if (checker & (1 << bitIndex)) {
            return 0;
        }
        checker |= (1 << bitIndex);
        str++;
    }
    return 1;
}

int main() {
    const char *testCases[] = {"abcdef", "aabcde", "xyz", "aabb", "programming"};
    int n = sizeof(testCases) / sizeof(testCases[0]);
    for (int i = 0; i < n; i++) {
        printf("\"%s\" -> %s\n", testCases[i],
               hasAllUniqueCharacters(testCases[i]) ? "All unique" : "Has duplicates");
    }
    return 0;
}

File: python/unique_character.py
def has_all_unique_characters(s):
    checker = 0
    for c in s:
        if not ('a' <= c <= 'z'):
            raise ValueError(f"Character '{c}' is not a lowercase letter (a-z).")
        bit_index = ord(c) - ord('a')
        if checker & (1 << bit_index):
            return False
        checker |= (1 << bit_index)
    return True


if __name__ == "__main__":
    test_cases = ["abcdef", "aabcde", "xyz", "aabb", "programming"]
    for input_str in test_cases:
        result = "All unique" if has_all_unique_characters(input_str) else "Has duplicates"
        print(f'"{input_str}" -> {result}')
